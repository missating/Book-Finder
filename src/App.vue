<template>
  <v-app>
    <v-toolbar app>
      <v-toolbar-title class="headline text-uppercase">
        <span class="font-weight-light">Book Finder</span>
      </v-toolbar-title>
    </v-toolbar>

    <v-content>
      <input-box></input-box>
      <router-view></router-view>
    </v-content>
  </v-app>
</template>

<script>
import InputBox from '@/components/InputBox.vue';

export default {
  name: 'App',
  components: {
    InputBox,
  },
  data() {
    return {};
  },
};
</script>
