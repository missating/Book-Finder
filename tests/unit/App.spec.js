import Vue from 'vue';
import Vuex from 'vuex';
import Vuetify from 'vuetify';
import { mount, createLocalVue } from '@vue/test-utils';
import App from '../../src/App';

Vue.use(Vuetify);
const localVue = createLocalVue();
localVue.use(Vuex);

describe('App.vue', () => {
  let store;

  beforeEach(() => {
    store = new Vuex.Store({
      state: {
        books: [],
        searchQuery: '',
        isFetching: false,
        apiBaseUrl: 'https://www.googleapis.com/books/v1/volumes',
      },
    });
  });

  it('should render correctly', () => {
    const wrapper = mount(App, {
      store,
      localVue,
    });
    expect(wrapper.element).toMatchSnapshot();
  });
});
