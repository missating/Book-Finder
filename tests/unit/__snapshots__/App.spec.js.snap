// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.vue should render correctly 1`] = `
<div
  class="application theme--light"
  data-app="true"
  id="app"
>
  <div
    class="application--wrap"
  >
    <nav
      class="v-toolbar v-toolbar--fixed theme--light"
      style="margin-top: 0px; padding-right: 0px; padding-left: 0px; transform: translateY(0px);"
    >
      <div
        class="v-toolbar__content"
        style="height: 64px;"
      >
        <div
          class="v-toolbar__title headline text-uppercase"
        >
          <span
            class="font-weight-light"
          >
            Book Finder
          </span>
        </div>
      </div>
    </nav>
     
    <main
      class="v-content"
      style="padding: 64px 0px 0px 0px;"
    >
      <div
        class="v-content__wrap"
      >
        <div
          class="container"
        >
          <form
            class="v-form"
            novalidate="novalidate"
          >
            <div
              class="container align-center justify-center"
            >
              <div
                class="layout"
              >
                <div
                  class="flex xs12"
                >
                  <div
                    class="v-input v-text-field theme--light"
                  >
                    <div
                      class="v-input__control"
                    >
                      <div
                        class="v-input__slot"
                      >
                        <div
                          class="v-text-field__slot"
                        >
                          <label
                            class="v-label theme--light"
                            for="input-box"
                            style="left: 0px; position: absolute;"
                          >
                            Search For Books by: Title, Author etc
                          </label>
                          <input
                            id="input-box"
                            required="required"
                            type="text"
                          />
                        </div>
                      </div>
                      <div
                        class="v-text-field__details"
                      >
                        <div
                          class="v-messages theme--light"
                        >
                          <div
                            class="v-messages__wrapper"
                          />
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                 
                <div
                  class="flex xs12 md4"
                >
                  <button
                    class="v-btn theme--light"
                    raised=""
                    type="button"
                  >
                    <div
                      class="v-btn__content"
                    >
                      <span>
                        Search
                      </span>
                    </div>
                  </button>
                </div>
              </div>
            </div>
          </form>
           
          <!---->
           
          <!---->
        </div>
         
        <router-view />
      </div>
    </main>
  </div>
</div>
`;
